src/ConversationList/ConversationListItem.vala

```diff
@@ -30,6 +30,16 @@ public class Mail.ConversationListItem : Gtk.ListBoxRow {
 
     public Camel.FolderThreadNode node { get; private set; }
 
+    public int64 timestamp {
+        get {
+            if (node.message.date_received == 0) {
+                // Sent messages do not have a date_received timestamp.
+                return node.message.date_sent;
+            }
```

Comment:
Small codestyle issue, put an empty (as in no indentation) new line between end of curly bracket and new code